### ISSUE TITLE
fitbit-asap doesn't force a connection

### ISSUE THREAD

**Issue by ds17f** [dillpixel/fitbit-asap#16]
I have a Google Pixel and a Fitbit Versa.

On the fitbit, a call to `peerSocket.send` is used as a means to wake the connection between the device and app.  If no send is generated, connectivity is left at the will of the device (I think).

The result was that messages were being queued, but not sent in a timely manner.  I addressed this by adding a ping method which I call on startup.  It looks like this: 

```
const pingConnection = (timeout = 1000) => {
  // make a ping method which
  // sleeps for a timeout
  // then sends a message
  const ping = () => {
    console.log("ping");
    setTimeout(() => {
      peerSocket.send("ping");
    }, timeout);
  };

  // add a listener to the error event
  // which will ping again when after an error
  peerSocket.addEventListener('error', () => {
    ping();
  });

  // send the first ping
  ping();
};
```

Effectively this will send pings every second until there's a successful connection at which point it will not resend.  This forces the connection to establish, though will have a battery penalty should the connection be unavailable.

Would you consider something like this for `fitbit-asap`?

**Comment by Artaud** [comment]
Big thanks for your contribution. I believe we should check with hexxeh whether the peerSocket behaviour is really such as you describe. With the big rewrite of the comm stack I'm not sure.

**Comment by ds17f** [comment]
Please do.  That's a good idea.  With that said, necessity is the mother of invention.  I implemented this in an app that's on the 4.0 SDK because my app wasn't connecting early enough.  Adding the explicit call seems to have had an effect.

This was recommended to me by "FitBit Jon" in the Discord chat.

With that said, we should definitely confirm my suspicion as I'm not 100% sure.

**Comment by daveyjones** [comment]
@ds17f A peer connection should be established automatically every time an app is started, and the connection should run continuously as long as the app is running. If this doesn't happen automatically, then something else in the comms chain may be failing.

Calling `send` before the peer socket is ready _shouldn't_ have any affect on the the speed at which the automatic connection is established. That being said, it might be worth measuring to confirm:

```js
const start = Date.now()
try {
  peerSocket.send("ping")
} catch (error) {
  console.log(error)
}
peerSocket.addEventListener("open", () => {
  console.log(Date.now() - start)
})
```

When you have some time, it would be useful to run this test on a real device a few times with and without the ping line (ideally on both Android and iOS, if possible). If there's a measurable difference, then we should definitely take advantage of it!
